src/AppInstallerCommonCore/MSStore.cpp

```diff
@@ -209,11 +235,15 @@ namespace AppInstaller::MSStore
             installOptions).get();
 
         // Check if we need to restart or cancel existing items.
-        if (RestartOrCancelExistingOperationIfNecessary(installItems, installManager, m_productId))
+        auto restartOrCancelResult = RestartOrCancelExistingOperationIfNecessary(installItems, installManager, m_productId);
+        if (FAILED(restartOrCancelResult))
         {
-            // Wait a bit
-            Sleep(500);
+            return restartOrCancelResult;
+        }

```

Comment:
Get rid of this and just us `RETURN_IF_FAILED` on the call to `RestartOrCancelExistingOperationIfNecessary`.